app/components/TimeUntilUnveil.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from "react";
+
+const getOrdinalSuffix = (day: number): string => {
+  const suffixes = ["th", "st", "nd", "rd"];
+  const value = day % 100;
+  return suffixes[(value - 20) % 10] || suffixes[value] || suffixes[0];
+};
+const formatUnveilDate = (date: Date): string => {
+  const day = date.getDate();
+  const month = date.toLocaleString("default", { month: "long" });
+  const year = date.getFullYear();
+  const hours = date.getHours();
+  const minutes = date.getMinutes();
+
+  const ordinalDay = `${day}${getOrdinalSuffix(day)}`;
+
+  const formattedTime = `${hours}:${minutes < 10 ? `0${minutes}` : minutes}`;
+
+  return `${ordinalDay} ${month} ${year} ${formattedTime}`;
+};
+
+const calculateTimeUntilUnveil = (unveilDate: Date): string => {
+  const now = new Date();
+  const timeDiff = unveilDate.getTime() - now.getTime();
+
+  if (timeDiff <= 0) {
+    return "Unveiled";
+  }
+
+  const days = Math.floor(timeDiff / (1000 * 3600 * 24));
+  const hours = Math.floor((timeDiff % (1000 * 3600 * 24)) / (1000 * 3600));
+  const minutes = Math.floor((timeDiff % (1000 * 3600)) / (1000 * 60));
+  const seconds = Math.floor((timeDiff % (1000 * 60)) / 1000);
+
+  const formattedHours = hours < 10 ? `0${hours}` : hours;
+  const formattedMinutes = minutes < 10 ? `0${minutes}` : minutes;
+  const formattedSeconds = seconds < 10 ? `0${seconds}` : seconds;
+
+  return `${days} Days - ${formattedHours}hr:${formattedMinutes}min:${formattedSeconds}sec`;
+};
+
```

Comment:
You will be provided the time as props in this format if the time is yet to come `{ isUnveiled: false, Days: X, Minutes: X, Seconds: X }`  and ` { isUnveiled: true, Days: 0, Minutes: 0, Seconds: 0}` if the time has passed

use that to show the rendered time instead of creating a function